### PR TITLE
python312Packages.nbconflux: fix broken build by replacing versioned code

### PR DESCRIPTION
**Pull request by sarahec**
## Description of changes

TL;dr Fixes a minor component  of `jupyter` that is blocking builds on aarch64-Darwin.

`python312Packages.nbconflux` fails to build due to its use of `SafeConfigParser` which was renamed to `ConfigParser` in Python 3.2. This comes from a versioned copy of `python312Packages.versioneer`. This results in builds failing on aarch64-Darwin as the package sources predate aarch64 by several years. Older systems got along just fine on the cached binary copy. 

Changes:
1. Replaces the vendorized copy of versioneer with a dependency. (Thank you @emilazy)
2. Moves the Python requirement from >= 2.7 to >= 3.2. 
3. Fixes the src.owner to account for the repo owners changing their org name several years back. It's pulling from the same source base as always. (After clearing the hash and updating it, it didn't change.)
4. Updates the build system to the current standard.

cc: @NixOS/darwin-maintainers
 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by sarahec** [comment]
Result of `nixpkgs-review pr 340615` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.nbconflux</li>
    <li>python311Packages.nbconflux.dist</li>
    <li>python312Packages.nbconflux</li>
    <li>python312Packages.nbconflux.dist</li>
  </ul>
</details>

**Review by emilazy — CHANGES_REQUESTED** [review]
The file you’re patching is a vendored copy of [Versioneer](https://github.com/python-versioneer/python-versioneer). I think we should simply de‐vendor it; if you grep the Nixpkgs tree for `versioneer` you’ll find instances of this pattern:

```nix
  postPatch = ''
    # Remove vendorized versioneer.py
    rm versioneer.py
  '';

  build-system = [
    setuptools
    versioneer
  ];
```

**Comment by sarahec** [comment]
> The file you’re patching is a vendored copy of [Versioneer](https://github.com/python-versioneer/python-versioneer). I think we should simply de‐vendor it; 

Done and tested.

**Comment by sarahec** [comment]
Result of `nixpkgs-review pr 340615` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.nbconflux</li>
    <li>python311Packages.nbconflux.dist</li>
    <li>python312Packages.nbconflux</li>
    <li>python312Packages.nbconflux.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/nbconflux/default.nix [hunk 1]
```diff
@@ -2,26 +2,33 @@
...
+  setuptools,
+  versioneer,
 }:
 
 buildPythonPackage rec {
   pname = "nbconflux";
   version = "0.7.0";
   format = "setuptools";
-  disabled = isPy27; # no longer compatible with python 2 urllib
+  disabled = pythonOlder "3.2"; # SafeConfigParser was deprecated here
```
Comment (emilazy) [inline pkgs/development/python-modules/nbconflux/default.nix:18]: This is now unnecessary and can be dropped, because `versioneer` encodes its requirements (well, it would be unnecessary in general because we don’t package a Python 3.2 implementation anyway, though).

LGTM other than that!

**Review by emilazy — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/nbconflux/default.nix (hunk 1 above)
Comment (sarahec) [reply pkgs/development/python-modules/nbconflux/default.nix]: Doooooone!

File: pkgs/development/python-modules/nbconflux/default.nix [hunk 2]
```diff
@@ -2,26 +2,31 @@
...
   requests,
   responses,
+  setuptools,
+  versioneer,
 }:
 
 buildPythonPackage rec {
   pname = "nbconflux";
   version = "0.7.0";
   format = "setuptools";
```
Comment (emilazy) [inline pkgs/development/python-modules/nbconflux/default.nix:16]: Since we’re using `build-system` definitions now, this should be the modern `pyproject = true;` to use the new Python build system, and `propagatedBuildInputs` should be renamed to `dependencies`.

**Review by emilazy — CHANGES_REQUESTED** [review]
Thanks! Sorry, one more nit that I really ought to have caught before…

File: pkgs/development/python-modules/nbconflux/default.nix (hunk 2 above)
Comment (sarahec) [reply pkgs/development/python-modules/nbconflux/default.nix:16]: Done. Will push momentarily.

**Review by emilazy — APPROVED** [review]
LGTM!

Result of `nixpkgs-review pr 340615` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.nbconflux</li>
    <li>python311Packages.nbconflux.dist</li>
    <li>python312Packages.nbconflux</li>
    <li>python312Packages.nbconflux.dist</li>
  </ul>
</details>
